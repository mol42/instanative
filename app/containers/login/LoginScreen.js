import React, { Component } from 'react'
import { View, Image, TextInput, ImageBackground } from 'react-native'
import Button from 'react-native-button';
import BaseScreen from "../BaseScreenRN";
import ErrorMessage from "../../components/ErrorMessage";
// redux
import { connect } from "react-redux";
import { tryAutoLogin, tryLogin, updateLoginForm } from "../../redux/auth/Actions";

const styles = {
  input : {
    flex: 1, 
    color: "#fff", 
    height: 50, 
    marginLeft: 50, 
    marginRight : 50, 
    padding: 10, 
    borderWidth : 0.5, 
    borderColor : "#eee"
  }
}

export class LoginScreen extends BaseScreen {

  static navigationOptions = { 
    header: null, 
    tabBarVisible: false 
  }
  isVisible = false;

  constructor(props) {
    super(props);
  }

  componentWillMount() {
    // this.props.tryAutoLogin();
  }

  /**
   * saga login işlemini tamamladığı zaman yeni verilerin redux veri ağacına
   * set edilmesini sağlar ve güncellenen redux veri ağacı değişimi
   * dinleyen komponentlere gönderir. O anda aktif ekran login olduğu için
   * login başarılı ise ana ekrana gideriz.
   * 
   */
  componentWillReceiveProps(newProps) {
    let {loginInProgress, loginCompleted, loginHasError, user} = newProps.auth;
    const {state} = this.props.navigation;

    if (!loginInProgress && loginCompleted && !loginHasError && user.id > 0 && !this.isVisible) {
      this.navigateToScreen('HomeScreen');
      this.isVisible = true;
    }
  }

  render() {
    let {loginCompleted, loginHasError, loginForm} = this.props.auth;
    let {email, password} = loginForm;

    return <ImageBackground imageStyle={{resizeMode: "stretch", width: null, height: null}} 
                            source={require("../../images/login-bg-2.jpg")} 
                            style={{flex: 1, backgroundColor : "black"}}>
              <View style={{flex: 1, alignItems : "center", justifyContent : "center"}}>
                <Image source={require("../../images/insta-logo.png")} style={{width: 125, height: 125}} />
                <Image source={require("../../images/Instagram_logo.svg.png")} style={{width: 120, height : 36, marginTop: 20}} />
              </View>
              <View style={{flex: 1}}>
                <View style={{height: 50}}>
                    <TextInput autoCapitalize={"none"} value={email} style={styles.input} onChangeText={this._onEmailChanged} />
                </View>
                <View style={{height: 50, marginTop: 20}}>
                    <TextInput autoCapitalize={"none"} secureTextEntry={true} value={password} style={styles.input} onChangeText={this._onPasswordChanged} />
                </View>
                <View style={{height: 50, marginTop: 20}}>
                  <Button
                    containerStyle={{justifyContent : "center", borderRadius : 5, backgroundColor : "white", height: 50, marginLeft: 50, marginRight : 50}}
                    style={{color : "#444"}}
                    onPress={this._onLoginButtonClicked}>
                    Login
                  </Button>
                </View>
              </View>
              {/*<ErrorMessage enableError={loginCompleted} showError={loginCompleted && loginHasError} />*/}
          </ImageBackground>
  }

  _onLoginButtonClicked = () => {
    let {email, password} = this.props.auth.loginForm;
    this.props.tryLogin(email, password);
  }

  _onEmailChanged = (newEmail) => {
    let {email, password} = this.props.auth.loginForm;
    this.props.updateLoginForm(newEmail, password);
  }

  _onPasswordChanged = (newPassword) => {
    let {email, password} = this.props.auth.loginForm;
    this.props.updateLoginForm(email, newPassword);
  }
}

function bindAction(dispatch) {
  return {
    tryAutoLogin : () => dispatch(tryAutoLogin()),
    tryLogin : (email, password) => dispatch(tryLogin(email, password)),
    updateLoginForm : (email, password) => dispatch(updateLoginForm(email, password))
  };
}

const mapStateToProps = state => ({
    auth : state.auth
});

export default connect(mapStateToProps, bindAction)(LoginScreen);
