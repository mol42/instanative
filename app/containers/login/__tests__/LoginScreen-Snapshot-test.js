jest.mock('native-toaster', () => () => {
  show : jest.fn()
});

import React from 'react';
import renderer from 'react-test-renderer';

import { LoginScreen } from '../LoginScreen';

it('renders a LoginScreen using Snapshots', () => {
  
  const _getComponent = (props) => {
    return (
      renderer.create(
        <LoginScreen
          auth={props}
          tryAutoLogin={jest.fn()}
        />
      )
    )
  };
  
  //let component = _getComponent({ loginForm : {} });
  //expect(component).toMatchSnapshot();

  let component = _getComponent({ loginForm : {email : "user1@mol42.com", password : "1234"} });
  expect(component).toMatchSnapshot();
});
