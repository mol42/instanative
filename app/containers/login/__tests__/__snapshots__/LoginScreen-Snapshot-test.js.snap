// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a LoginScreen using Snapshots 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "flex": 1,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../app/images/login-bg-2.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        Object {
          "height": null,
          "resizeMode": "stretch",
          "width": null,
        },
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../app/images/insta-logo.png",
        }
      }
      style={
        Object {
          "height": 125,
          "width": 125,
        }
      }
    />
    <Image
      source={
        Object {
          "testUri": "../../../app/images/Instagram_logo.svg.png",
        }
      }
      style={
        Object {
          "height": 36,
          "marginTop": 20,
          "width": 120,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "height": 50,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        style={
          Object {
            "borderColor": "#eee",
            "borderWidth": 0.5,
            "color": "#fff",
            "flex": 1,
            "height": 50,
            "marginLeft": 50,
            "marginRight": 50,
            "padding": 10,
          }
        }
        value="user1@mol42.com"
      />
    </View>
    <View
      style={
        Object {
          "height": 50,
          "marginTop": 20,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        secureTextEntry={true}
        style={
          Object {
            "borderColor": "#eee",
            "borderWidth": 0.5,
            "color": "#fff",
            "flex": 1,
            "height": 50,
            "marginLeft": 50,
            "marginRight": 50,
            "padding": 10,
          }
        }
        value="1234"
      />
    </View>
    <View
      style={
        Object {
          "height": 50,
          "marginTop": 20,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits="button"
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 5,
            "height": 50,
            "justifyContent": "center",
            "marginLeft": 50,
            "marginRight": 50,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007aff",
                "fontSize": 17,
                "fontWeight": "500",
                "textAlign": "center",
              },
              null,
              Object {
                "color": "#444",
              },
              null,
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</View>
`;
